packages/main/src/MenuItem.ts

```diff
@@ -118,58 +143,136 @@ class MenuItem extends UI5Element {
 	@property({ type: String })
 	tooltip!: string;
 
-	/**
-	 * Indicates whether any of the element siblings have children items.
-	 */
-	@property({ type: Boolean, noAttribute: true })
-	_siblingsWithChildren!: boolean;
-
 	/**
 	 * Indicates whether any of the element siblings have icon.
 	 */
 	@property({ type: Boolean, noAttribute: true })
 	_siblingsWithIcon!: boolean;
 
-	/**
-	 * Defines whether the submenu closing must be prevented
-	 */
-	@property({ type: Boolean, noAttribute: true })
-	_preventSubMenuClose!: boolean;
-
-	/**
-	 * Stores Menu object with submenu items
-	 */
-	@property({ type: Object, defaultValue: undefined })
-	_subMenu?: Menu;
-
 	/**
 	 * Defines the items of this component.
 	 * @public
 	 */
 	@slot({ "default": true, type: HTMLElement, invalidateOnChildChange: true })
 	items!: Array<MenuItem>;
 
-	get hasSubmenu() {
-		return !!(this.items.length || this.loading);
+	get placement(): `${PopoverPlacement}` {
+		return this.isRtl ? "Start" : "End";
 	}
 
-	get hasDummyIcon() {
-		return this._siblingsWithIcon && !this.icon;
+	get isRtl() {
+		return this.effectiveDir === "rtl";
+	}
+
+	get hasSubmenu() {
+		return !!(this.items.length || this.loading);
 	}
 
-	get subMenuOpened() {
-		return !!this._subMenu?._popover?.isOpen();
+	get hasIcon() {
+		return !!this.icon;
 	}
 
-	get _additionalText() {
-		return this.hasSubmenu ? "" : this.additionalText;
+	get isSubMenuOpen() {
+		return this._popover?.isOpen();
 	}
 
 	get ariaLabelledByText() {
 		return `${this.text} ${this.accessibleName}`.trim();
 	}
+
+	get menuHeaderTextPhone() {
+		return this.text;
+	}
+
+	get isPhone() {
+		return isPhone();
+	}
+
+	get labelBack() {
+		return MenuItem.i18nBundle.getText(MENU_BACK_BUTTON_ARIA_LABEL);
+	}
+
+	get labelClose() {
+		return MenuItem.i18nBundle.getText(MENU_CLOSE_BUTTON_ARIA_LABEL);
+	}
+
+	onBeforeRendering() {
+		const siblingsWithIcon = this.items.some(item => !!item.icon);
+
+		if (siblingsWithIcon) {
```

Comment:
what if there is one sibling with icon, and at some moment the icon is removed? Nobody will reset other items' _siblingsWithIcon property... Maybe will be better to do:

```
this.items.forEach(item => {
	item._siblingsWithIcon = siblingsWithIcon;
});
```

without the prior IF?